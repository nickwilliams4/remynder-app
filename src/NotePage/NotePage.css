.newRemynder {
  text-decoration: underline;
}


ul {
  list-style: none;
}