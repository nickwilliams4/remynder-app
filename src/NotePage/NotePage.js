import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import './NotePage.css'
import Note from '../Note/Note'
import ApiContext from '../ApiContext'
import config from '../config'
import TokenService from '../services/token-service'

export default class NotePage extends Component {
  static contextType = ApiContext

  componentDidMount() {
    fetch(`${config.API_ENDPOINT}/notes`, {
      headers: {
        'Content-Type': 'application/json',
        'authorization': `bearer ${TokenService.getAuthToken()}`
      },
    })
    
      .then((notesRes) => {
        if (!notesRes.ok)
          return notesRes.json().then(e => Promise.reject(e));
        return notesRes.json();
      })
      .then((notes) => {
        this.context.setNotes( notes );
      })
      .catch(error => {
        console.error({ error });
      });
  }

  render() {
    const { notes = [] } = this.context
    return (
      <main role="main">
        <header role="banner">
          <h1>Your Remynders</h1>
        </header>
        <div>
          
        </div>
        <section>
          <header>
            <h2 className='newRemynder'>Create New Remynder</h2>
            <Link to='/AddNote'>
              <button>New Remynder</button>
            </Link>
          </header>
        </section>
        <section className='NoteList'>
          <ul>
            {notes.map(note =>
              <li key={note.id}>
                <Note
                  id={note.id}
                  title={note.title}
                  content={note.content}
                  created={note.created}
                  remynder={note.remynder}
                />
              </li>
            )}
          </ul>
        </section>
      </main>
    )
  }
}