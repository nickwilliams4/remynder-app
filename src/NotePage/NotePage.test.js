import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NotePage from './NotePage'

describe(`NotePage component`, () => {
  const props = {
    id: 'a',
    name: 'test-class-name',
  }

  it('renders the NotePage', () => {
    const wrapper = shallow(<NotePage {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})