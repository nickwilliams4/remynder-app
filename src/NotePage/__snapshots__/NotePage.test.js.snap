// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotePage component renders the NotePage 1`] = `
<main
  role="main"
>
  <header
    role="banner"
  >
    <h1>
      Your Remynders
    </h1>
  </header>
  <div />
  <section>
    <header>
      <h2
        className="newRemynder"
      >
        Create New Remynder
      </h2>
      <Link
        to="/AddNote"
      >
        <button>
          New Remynder
        </button>
      </Link>
    </header>
  </section>
  <section
    className="NoteList"
  >
    <ul />
  </section>
</main>
`;
